src/main/java/com/ericsson/ei/controller/FailedNotificationControllerImpl.java

```diff
@@ -27,60 +31,83 @@
             final HttpServletRequest httpRequest) {
-        ObjectMapper mapper = new ObjectMapper();
-        QueryResponse queryResponse = new QueryResponse();
-        QueryResponseEntity queryResponseEntity = new QueryResponseEntity();
-        try {
-            List<String> response = processMissedNotification.processQueryMissedNotification(subscriptionName);
-            if (!response.isEmpty()) {
-                queryResponseEntity = mapper.readValue(response.get(0), QueryResponseEntity.class);
-            }
-            queryResponse.setQueryResponseEntity(queryResponseEntity);
-            LOGGER.debug("The response is : {}", response.toString());
-            if (processMissedNotification.deleteMissedNotification(subscriptionName)) {
-                LOGGER.debug("Failed notification for subscription {} was successfully removed from database",
-                        subscriptionName);
-             }
-            return new ResponseEntity<>(queryResponse, HttpStatus.OK);
-        } catch (Exception e) {
-            String errorMessage = "Failed to extract the data from the failed notification object based on subscription name "
-                    + subscriptionName + ".";
-            LOGGER.error(errorMessage, e);
+        List<String> subscriptionNameList = Arrays.asList(subscriptionNames.split(","));
+
+        JSONObject response = fetchFailedNotifications(subscriptionNameList);
+        JSONArray foundArray = response.getJSONArray(FOUND_NOTIFICATIONS);
+        JSONArray notFoundArray = response.getJSONArray(NOT_FOUND_NOTIFICATIONS);
+        HttpStatus httpStatus;
+        if (foundArray.length() > 0) {
+            httpStatus = HttpStatus.OK;
+        } else {
+            httpStatus = HttpStatus.NOT_FOUND;
+        }
+        if (httpStatus == HttpStatus.NOT_FOUND) {
+            String errorMessage = "Failed to fetch failed notifications for subscriptions:\n"
+                    + notFoundArray.toString();
             String errorJsonAsString = ResponseMessage.createJsonMessage(errorMessage);
-            return new ResponseEntity<>(errorJsonAsString, HttpStatus.INTERNAL_SERVER_ERROR);
+            return new ResponseEntity<>(errorJsonAsString, httpStatus);
+        }
+        return new ResponseEntity<>(response.toString(), httpStatus);
+    }
+
+    private JSONObject fetchFailedNotifications(List<String> subscriptionNameList) {
+        JSONObject response = new JSONObject();
+        JSONArray foundArray = new JSONArray();
+        JSONArray notFoundArray = new JSONArray();
+
+        for (String name : subscriptionNameList) {
+            List<String> notifications = null;
+            try {
+                notifications = processFailedNotification.processQueryFailedNotification(name);
+            } catch (NoSuchElementException e) {
+                LOGGER.error("Failed to fetch failed notification for subscription {}", name, e);
```

Comment:
Maybe this should not be "error" since this technically is not an error. So info would maybe make more sense or what do you think?

Reply:
Yea, I think you are right. There is nothing really alarming about not having found any notifications.

Reply:
I'll rephrase the sentence to No failed notifications found as well. "Failed" sounds too harsh.

Reply:
I actually didn't need to log anything specific here since the error that I throw already contains the information. I'll leave it blank.